docs/data/osdf/install-cache-container.md

```diff
@@ -0,0 +1,347 @@
+* __Service requirements:__
+  * A cache serving the OSDF federation as a regional cache should have at least:
+    * 8 cores
+    * 40 Gbps connectivity
+    * 50-200 TB of NVMe disk for the cache partition; you may distribute the disk, e.g., by using an NVMe-backed Ceph pool,
+        if you cannot fit that much disk into a single chassis
+    * 24 GB of RAM
+  * A cache being used to serve data from the OSDF to a single site should have at least:
+    * 8 cores
+    * 40 Gbps connectivity
+    * 2 TB of NVMe disk for the cache partition
+    * 24 GB of RAM
+
+
+!!! note "Host certificates"
+    Caches are accessed by users through browsers, meaning caches need a certificate from a CA acceptable to a standard browser.
+    Examples include [Let's Encrypt](../../security/host-certs/lets-encrypt.md) or the InCommon RSA CA.
+    Caches without a valid certificate for the browser cannot be added to the OSDF.
+    Note that, unlike legacy grid software, the public certificate file will need to contain the "full chain", including any
+    intermediate CAs (if you're unsure about your setup, try accessing your cache from your browser).
+
+    The certificate chain and key should be mounted into the following locations:
+
+    * **Host Certificate Chain**: `/etc/pelican/certificates/tls.crt`
+    * **Host Key**: `/etc/pelican/certificates/tls.key`
+
+
+!!! note "osdf-cache 7.12"
+    This document covers versions 7.12 and later of the `osdf-cache` container image.
+
+!!! note "root"
+    The paths used as examples on this page (e.g., `/etc/pelican`) require root to edit;
+    if you do not have root on the host, modify the directories to a path you do have access to.
+
+
+Configuring the Cache Server
+----------------------------
+
+The preferred method of configuring a Pelican-based OSDF cache is via a YAML file that will be 
+mounted into the container into the `/etc/pelican/config.d` directory.
+This document will use `/etc/pelican/cache-config.yaml` as the name of the configuration file outside of the container.
+
+Create `/etc/pelican/cache-config.yaml` with the following contents:
+```file
+Cache:
+  Port: 8443
+# XRootD:
+#   Sitename: <RESOURCE NAME REGISTERED WITH OSG>
+```
+You will uncomment and fill in the value for `XRootD.Sitename` in a later step, after registration is complete.
+
+
+Preparing for Initial Startup
+-----------------------------
+
+1.  The cache identifies itself to the federation via public key authentication;
+before starting the cache for the first time, it is recommended to generate a keypair.
+Download the Pelican client from <https://github.com/PelicanPlatform/pelican/releases> and run:
+
+        :::console
+        root@host$ cd /etc/pelican
+        root@host$ pelican generate keygen
+
+
+    The newly created files, `issuer.jwk` and `issuer-pub.jwks` are the private and public keys, respectively.
+
+1.  **Save these files**; if you lose the `issuer.jwk`, your cache will need to be re-approved.
```

Comment:
We already know they're running `docker` so maybe just have them run the relevant command via container?

Reply:
Since you actually have to get the generated files out, you'd have to do some bind-mounting.  Also the image contains an ENTRYPOINT so you have to override that.  In 7.12, the command would be something like

```
docker run -v /etc/pelican:/pelican \
  --entrypoint /usr/local/bin/osdf \
  hub.opensciencegrid.org/pelican_platform/osdf-cache:latest \
  generate keygen
```
(this is untested); in 7.11 the command is
```
docker run -v /etc/pelican:/out \
  --entrypoint /pelican/osdf \
  hub.opensciencegrid.org/pelican_platform/osdf-cache:latest \
  generate keygen --private-key /out/issuer.jwk --public-key /out/issuer-pub.jwks
```
both of which I think are more complicated than downloading the client and running the command locally.